src/Shared/PackageManagers/NPMProjectManager.cs

```diff
@@ -180,8 +317,20 @@ public List<Version> GetVersions(JsonDocument contentJSON)
         /// <returns>
         ///     A dictionary, mapping each possible repo source entry to its probability/empty dictionary
         /// </returns>
-        protected async override Task<Dictionary<PackageURL, double>> PackageMetadataSearch(PackageURL purl,
+
+        protected async override Task<Dictionary<PackageURL, double>> SearchRepoUrlsInPackageMetadata(PackageURL purl,
             string metadata)
+        {
+            if (string.IsNullOrEmpty(metadata))
+            {
+                return new Dictionary<PackageURL, double>();
+            }
+            JsonDocument contentJSON = JsonDocument.Parse(metadata);
+            return await SearchRepoUrlsInPackageMetadata(purl, contentJSON);
+        }
+
+        protected async Task<Dictionary<PackageURL, double>> SearchRepoUrlsInPackageMetadata(PackageURL purl,
+            JsonDocument contentJSON)
         {
             var mapping = new Dictionary<PackageURL, double>();
             if (purl?.Name is string purlName && (purlName.StartsWith('_') || npm_internal_modules.Contains(purlName)))
```

Comment:
purl is non-null here because this is a protected API.

```suggestion
            if ((purl.Name.StartsWith('_') || npm_internal_modules.Contains(purl.Name)))
```